src/suffix_array/sais.rs

```diff
@@ -365,62 +387,96 @@ mod tests {
     }
 
     #[test]
-    #[should_panic(expected = "expected:")]
+    #[should_panic]
     fn test_sais_no_trailing_zero() {
         let text = "nozero".to_string().into_bytes();
         let converter = RangeConverter::new(b'a', b'z');
         build_suffix_array(&text, &converter);
     }
 
+    #[test]
+    #[should_panic]
+    fn test_sais_too_many_trailing_zero() {
+        let text = "toomanyzeros\0\0".to_string().into_bytes();
```

Comment:
Unfortunately, I found it difficult to support more than one trailing zeros in a given text. For now let me treat such text as unexpected input.